exercise-book/src/tcp-server.md

```diff
@@ -0,0 +1,85 @@
+<!-- markdownlint-disable MD033 -->
+# Interactive TCP Echo Server
+
+In this exercise, we will make a simple TCP "echo" server using APIs in Rust's Standard Library.
+
+Here's how an interaction with it would look like from a client point of view. You connect to it using `nc`, for example:
```

Comment:
How do I do `nc` on windows? Also, does this work with gnu and bsd `nc`? They have annoyingly different behaviour.

Reply:
We ship a TCP client in the tree for people without `nc`.

Reply:
I've been running these examples for more than a year, no issues with `nc` on macOS or Linux.

On Windows almost all participants either run `nc` from WSL or use Linux servers / vis to run these exercises. There's also a Windows version of it from `Nmap` project. I've added a troubleshooting note about all that to the first TCP exercise.